### PR TITLE
pkg/controller/node: add tests for multiple pools

### PR DESCRIPTION
**Pull request by runcom**
**- What I did**

Add tests for `getNodesForPool` - the e2e version of this is going to QE directly for now but I'm hoping we can add it to our suite eventually.

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: pkg/controller/node/node_controller_test.go [hunk 1]
```diff
@@ -325,6 +380,23 @@ func TestGetPoolForNode(t *testing.T) {
 
 func intStrPtr(obj intstr.IntOrString) *intstr.IntOrString { return &obj }
 
+func newMixedNodeSet(setlen int, roles ...map[string]string) []*corev1.Node {
```
Comment (kikisdeliveryservice) [inline pkg/controller/node/node_controller_test.go]: could we add a 1 liner comment about what these new helpers are?

File: pkg/controller/node/node_controller_test.go [hunk 2]
```diff
@@ -213,6 +214,60 @@ func (f *fixture) expectPatchNodeAction(node *corev1.Node, patch []byte) {
 	f.kubeactions = append(f.kubeactions, core.NewPatchAction(schema.GroupVersionResource{Resource: "nodes"}, node.Namespace, node.Name, types.MergePatchType, patch))
 }
 
+func TestGetNodesForPool(t *testing.T) {
```
Comment (kikisdeliveryservice) [inline pkg/controller/node/node_controller_test.go]: 1 liner comment for this?

File: pkg/controller/node/node_controller_test.go (hunk 1 above)
Comment (runcom) [reply pkg/controller/node/node_controller_test.go]: yup I'll add it

File: pkg/controller/node/node_controller_test.go (hunk 2 above)
Comment (runcom) [reply pkg/controller/node/node_controller_test.go]: This is testing the `getNodesForPool` function - following how `Test...` in golang are written

Comment (kikisdeliveryservice) [reply pkg/controller/node/node_controller_test.go]: ah sorry, still drinking coffee

File: pkg/controller/node/node_controller_test.go (hunk 1 above)
Comment (kikisdeliveryservice) [reply pkg/controller/node/node_controller_test.go]: thanks!

Comment (runcom) [reply pkg/controller/node/node_controller_test.go]: fixed!

**Pull request merged**